SOURCES/etc/xapi.d/plugins/ipmitool.py

```diff
@@ -0,0 +1,142 @@
+        if "Could not open device" in e.stderr:
+            return (
+                False,
+                "IPMI device not found. Ensure the IPMI module is loaded and that your system supports IPMI.",
+            )
+        else:
+            raise e
+    else:
+        return True, output
+
+
+@error_wrapped
+def sensor_data(_session, _args):
+    sensor_data = []
+    try:
+        success, output = run_ipmitool(["sdr", "list"])
+    except ProcessException as e:
+        raise e
+
+    if not success:
+        return json.dumps(output)
+
+    for line in output["stdout"].splitlines():
+        if not line:
+            continue
+        sensor_fields = line.split("|")
+        sensor_data.append(
+            {
+                "name": sensor_fields[0].strip(),
+                "value": sensor_fields[1].strip(),
+                "event": sensor_fields[2].strip(),
+            }
+        )
+
+    return json.dumps(sensor_data)
+
+
+@error_wrapped
+def sensor_info(_session, args):
+    sensors_info = []
+    sensors = args.get("sensors")
+
+    if not sensors:
+        return "{}"
+
+    for sensor in sensors.split(","):
+        sensor = sensor.strip()
+        info = []
+        try:
+            success, output = run_ipmitool(["sdr", "get", sensor])
+        except ProcessException as e:
+            raise e
+
+        if not success:
+            return json.dumps(output)
+
+        for line in output["stdout"].splitlines():
+            if ":" not in line:
+                continue
+            name, value = line.split(":", 1)
+            info.append(
+                {
+                    "name": name.strip(),
+                    "value": value.strip(),
+                }
+            )
+
+        sensors_info.append(
+            {
+                "name": sensor,
+                "info": info,
+            }
+        )
+
+    return json.dumps(sensors_info)
+
+
+@error_wrapped
+def ipmi_lan(_session, _args):
+    lan_info = []
+    wanted = [
+        "IP Address",
+        "Subnet Mask",
+        "MAC Address",
+        "BMC ARP Control",
+        "Default Gateway IP",
+        "802.1q VLAN",
+        "RMCP+ Cipher Suites",
+    ]
+
+    try:
+        success, output = run_ipmitool(["lan", "print"])
+    except ProcessException as e:
+        raise e
+
+    if not success:
+        return json.dumps(output)
+
+    for line in output["stdout"].splitlines():
+        if any(word in line for word in wanted):
+            name, value = line.split(":", 1)
+            lan_info.append(
+                {
+                    "name": name.strip(),
+                    "value": value.strip(),
+                }
+            )
+
+    return json.dumps(lan_info)
+
+
+_LOGGER = configure_logging("ipmitool-xapi-plugin")
+if __name__ == "__main__":
+    XenAPIPlugin.dispatch(
+        {
+            "get_all_sensors": sensor_data,
+            "get_sensor": sensor_info,
+            "get_ipmi_lan": ipmi_lan,
+        }
+    )
```

Comment:
```suggestion
    XenAPIPlugin.dispatch({
        "get_all_sensors": sensor_data,
        "get_sensor": sensor_info,
        "get_ipmi_lan": ipmi_lan,
    })
```

I guess you have an auto-format plugin. :)

Reply:
Yes I'm using [black](https://black.readthedocs.io/en/stable/index.html)

@Wescoeur What should I use to have your indentation because I prefer to use something automatic as I don't really care myself with indentation ;)

Reply:
I guess that we can use the future style: https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#improved-multiline-dictionary-and-list-indentation-for-sole-function-parameter

I had in mind to use black on smapi v3 but we could directly use it on this repo first in a single commit + with a check on the github workflow.

Reply:
Also notice that this file is a new one. So it won't be part of the first single commit. Also I'm not sure how to use the future style? Is it a parameter to pass to `black` command line?

Reply:
It looks like as you suggest `black --unstable` is doing the job :)